interfaces/builtin/unity8.go

```diff
@@ -40,6 +128,26 @@ func (iface *Unity8Interface) PermanentPlugSnippet(plug *interfaces.Plug, securi
 }
 
 func (iface *Unity8Interface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		oldTags := []byte("###SLOT_SECURITY_TAGS###")
+		newTags := slotAppLabelExpr(slot)
+		snippet := bytes.Replace([]byte(unity8ConnectedPlugAppArmor), oldTags, newTags, -1)
+
+		// FIXME: Until we decide whether unity8 is going to use Snappy-style
+		//        appIDs (snap.NAME.COMMAND) or Touch-style ones
+		//        (NAME_COMMAND_REVISION), we'll use a simpler *NAME* glob
+		//        here. This glob does not provide isolation because 'foo' can
+		//        access 'barfoobaz'. The base declaration restriction cannot
+		//        be lifted until this is properly mediated.
+		appidsOld := []byte("###PLUG_DBUS_APPIDS###")
+		appidsNew := fmt.Sprintf("*%s*", dbus.SafePath(plug.Snap.Name()))
```

Comment:
@jdstrand Your sentiment matches mine pretty much. Our only disagreement lies here:

> I agreed to the current approach with the understanding that the broken rules would be incompatibly removed before the base declaration was updated for general use,

Reality tends to be unforgiving in those cases. Code easily gets shipped in production which such  caveats, and then we're left with broken code that we can't drop, and we wouldn't even have the ability to statically analyze which snaps would actually be broken or not, because good and bad would look the same.

@mikix Thanks for that!